controllers/tools.js

```diff
@@ -231,31 +237,253 @@ const showTools = (req, res) => {
+    });
+};
 
-router.get('/search' , showTools);
+const getCreateHandlerCtl = (req, res, next) => {
+    // TODO validation/sanitization
 
-router.patch('/:id', getUpdateHandler('ltitools'));
-router.get('/:id', getDetailHandler('ltitools'));
-router.delete('/:id', getDeleteHandler('ltitools'));
-router.post('/', getCreateHandler('ltitools'));
-router.all('/', showTools);
+    if(req.body.config.type === 'oauth2') {
+        api(req, { version: HYDRA_VERSION }).post('/oauth2/clients/', {
+            json: mapToolToOauthClient(req.body, true)
+        }).then(response => {
+            req.body.config.clientId = response.client_id;
+            createCtlTool(req, next);
+        }).catch(err => {
+            next(err);
+        });
+    } else {
+        createCtlTool(req, next);
+    }
+};
+
+const showToolsCtl = (req, res) => {
+    const itemsPerPage = (req.query.limit || 10);
+    const currentPage = parseInt(req.query.p) || 1;
+
+    let sortOrder;
+    let sortBy;
+    if (req.query.sort) {
+        if (req.query.sort.startsWith('-')) {
+            sortOrder = 'desc';
+            sortBy = req.query.sort.substring(1);
+        } else {
+            sortOrder = 'asc';
+            sortBy = req.query.sort;
+        }
+
+        if(sortBy === '_id') {
+            sortBy = 'id';
+        } else if(sortBy === 'undefined') {
+            sortBy = undefined;
+        }
+    }
+
+    api(req, { version: 'v3' }).get('/tools', {
+        json: {
+            name: req.query.q,
+            limit: itemsPerPage,
+            skip: itemsPerPage * (currentPage - 1),
+            sortOrder,
+            sortBy,
+        },
+    }).then((tools) => {
+        const body = tools.data.map(item => {
+            return [
+                item.id || "",
+                item.name || "",
+                item.config.clientId || "",
+                getTableActions({ _id: item.id }, '/tools/')
+            ];
+        });
+
+        let sortQuery = '';
+        if (req.query.sort) {
+            sortQuery = '&sort=' + req.query.sort;
+        }
+
+        let limitQuery = '';
+        if (req.query.limit) {
+            limitQuery = '&limit=' + req.query.limit;
+        }
+
+        const pagination = {
+            currentPage,
+            numPages: Math.ceil(tools.total / itemsPerPage),
+            baseUrl: '/tools/?p={{page}}' + sortQuery + limitQuery
+        };
 
-router.get('/', function (req, res, next) {
-    api(req).get('/ltitools/').then(ltitools => {
         res.render('tools/tools', {
             title: 'Tools',
+            head,
+            body,
+            pagination,
             user: res.locals.currentUser,
-            tools: ltitools.data,
-            themeTitle: process.env.SC_NAV_TITLE || 'Schul-Cloud'
+            limit: true,
+            themeTitle: process.env.SC_NAV_TITLE || 'Schul-Cloud',
         });
     });
-});
+};
+
+
+// secure routes
+router.use(authHelper.authChecker);
+
+if(LTI_VERSION === 'v3') {
```

Comment:
Can you rename this variable to something like USE_CTL or NEW_TOOLS or something like this. At the first moment i thought it has something to do with the version of lti tools